modular_skyrat/modules/borg_buffs/code/robot.dm

```diff
@@ -116,3 +116,78 @@
 		tool_behaviour = TOOL_KNIFE
 		to_chat(user, span_notice("You attach the knife bit to [src]."))
 		icon_state = "knife_screw_cyborg"
+
+/obj/item/inducer/cyborg
+	name = "Cyborg Inducer"
+	desc = "A tool for inductively charging internal power cells using the battery of a cyborg"
+	powertransfer = 250
+
+
+
+/obj/item/inducer/cyborg/attackby(obj/item/Weapon, mob/user)
+	if(Weapon.tool_behaviour == TOOL_SCREWDRIVER)
+		return
+	if(istype(Weapon, /obj/item/stock_parts/cell))
+		return
+
+/obj/item/inducer/cyborg/recharge(atom/movable/target_atom, mob/user)
+	if(!iscyborg(user))
+		return
+	var/mob/living/silicon/robot/borg_user = user
+	cell = borg_user.cell
+	if(!isturf(target_atom) && user.loc == target_atom)
+		return FALSE
+	if(recharging)
+		return TRUE
+	else
+		recharging = TRUE
+	var/obj/item/stock_parts/cell/target_cell = target_atom.get_cell()
+	var/obj/target_object
+	var/coefficient = 1
+	if(istype(target_atom, /obj/item/gun/energy))
+		to_chat(user, span_alert("Error unable to interface with device."))
+		return FALSE
+	if(istype(target_atom, /obj/item/clothing/suit/space))
+		to_chat(user, span_alert("Error unable to interface with device."))
+		return FALSE
+	if(cell.charge <= 1000 ) // Cyborg charge safety. Prevents a borg from inducing themself to death.
+		to_chat(user, span_alert("Unable to charge device. User battery safety engaged."))
+		return
+	if(istype(target_atom, /obj))
+		target_object = target_atom
+	if(target_cell)
+		var/done_any = FALSE
+		if(target_cell.charge >= target_cell.maxcharge)
+			to_chat(user, span_notice("[target_atom] is fully charged!"))
+			recharging = FALSE
+			return TRUE
+		user.visible_message(span_notice("[user] starts recharging [target_atom] with [src]."), span_notice("You start recharging [target_atom] with [src]."))
+		while(target_cell.charge < target_cell.maxcharge)
+			if(do_after(user, 10, target = user) && cell.charge)
+				done_any = TRUE
+				induce(target_cell, coefficient)
+				do_sparks(1, FALSE, target_atom)
+				if(target_object)
+					target_object.update_appearance()
+			else
+				break
+		if(done_any) // Only show a message if we succeeded at least once
+			user.visible_message(span_notice("[user] recharged [target_atom]!"), span_notice("You recharged [target_atom]!"))
+		recharging = FALSE
+		return TRUE
+	recharging = FALSE
```

Comment:
You're not returning anything at the end here, so it's going to be basically like `return null`, you might want to make it explicit using `return FALSE` there, so we know it's intended.